common/js/solid.js

```diff
@@ -0,0 +1,440 @@
+      this.passwordStrengthMeter.classList,
+      this.tokenize(this.getStrengthLevelProgressClass())
+    )
+  }
+
+  PasswordValidator.prototype.setPasswordFeedback = function () {
+    var feedback = this.getFeedbackFromLevel()
+    this.updateStrengthMeter()
+    this.displayPasswordErrors()
+    this.setFeedbackForField(feedback, this.passwordField)
+  }
+
+  /**
+   * Update the repeated password feedback icon and color
+   */
+  PasswordValidator.prototype.updateRepeatedPasswordFeedback = function () {
+    var feedback = this.checkPasswordFieldsEquality() ? this.FEEDBACK_SUCCESS : this.FEEDBACK_ERROR
+    this.setFeedbackForField(feedback, this.repeatedPasswordField)
+  }
+
+  /**
+   * Display the given feedback on the field
+   * @param {string} feedback success|error|warning
+   * @param {HTMLElement} field
+   */
+  PasswordValidator.prototype.setFeedbackForField = function (feedback, field) {
+    var formGroup = this.getFormGroupElementForField(field)
+    var visualFeedback = this.getFeedbackElementForField(field)
+
+    this.resetValidation(formGroup)
+    this.resetFeedbackIcon(visualFeedback)
+
+    visualFeedback.classList.remove('hidden')
+
+    visualFeedback.classList
+      .add
+      .apply(
+        visualFeedback.classList,
+        this.tokenize(this.getFeedbackIconClass(feedback))
+      )
+
+    formGroup.classList
+      .add
+      .apply(
+        formGroup.classList,
+        this.tokenize(this.getValidationClass(feedback))
+      )
+  }
+
+  /**
+   * Password Strength Helpers
+   */
+
+  /**
+   * Get OWASP feedback on the given password. Returns false if the password is empty
+   * @param password
+   * @returns {object|false}
+   */
+  PasswordValidator.prototype.getPasswordStrength = function (password) {
+    if (password === '') {
+      return false
+    }
+    return owaspPasswordStrengthTest.test(password)
+  }
+
+  /**
+   * Get the password strength level based on password strength feedback object given by OWASP
+   * @param passwordStrength
+   * @returns {number}
+   */
+  PasswordValidator.prototype.getStrengthLevel = function (passwordStrength) {
+    if (passwordStrength === false) {
+      return 0
+    }
+    if (passwordStrength.requiredTestErrors.length !== 0) {
+      return 1
+    }
+
+    if (passwordStrength.strong === false) {
+      return 2
+    }
+
+    if (passwordStrength.isPassphrase === false || passwordStrength.optionalTestErrors.length !== 0) {
+      return 3
+    }
+
+    return 4
+  }
+
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP = []
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[0] = PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[1] = PasswordValidator.prototype.FEEDBACK_ERROR
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[2] = PasswordValidator.prototype.FEEDBACK_WARNING
+  PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[3] = PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP[4] = PasswordValidator.prototype.FEEDBACK_SUCCESS
+
```

Comment:
Just this would do:

```
PasswordValidator.prototype.LEVEL_TO_FEEDBACK_MAP = [
   FEEDBACK_ERROR, FEEDBACK_ERROR, FEEDBACK_WARNING, FEEDBACK_SUCCESS
]
```

if we additionally expose those as variables. Maybe they don't even need to be on the prototype.

Reply:
Fixed